### PR TITLE
REF: WalletsList TSX

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/5cspbvh6r7afziu5vsdkhs3agm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1184)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: helpers/getIsLargeScreen.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Dimensions, Platform } from 'react-native';
+import { isTablet } from 'react-native-device-info';
+import { isDesktop } from '../blue_modules/environment';
+
+// Helper function to determine if the screen is large
+export const getIsLargeScreen = (): boolean => {
+  const width = Dimensions.get('screen').width;
+  return Platform.OS === 'android' ? isTablet() : (width >= width / 2 && isTablet()) || isDesktop;
```
Comment (Overtorment) [inline helpers/getIsLargeScreen.ts:8]: wtf is `width >= width / 2`
any positive number satisfies this

Comment (Overtorment) [reply helpers/getIsLargeScreen.ts:8]: also, why  the difference between android & ios? leave a comment explanation

File: screen/wallets/WalletsList.tsx [hunk 2]
```diff
@@ -34,12 +33,89 @@ import presentAlert from '../../components/Alert';
...
-const WalletsList = () => {
-  const walletsCarousel = useRef();
-  const currentWalletIndex = useRef(0);
+type SectionData = {
+  key: string;
+  data: Transaction[] | string[];
+};
+
+enum ActionTypes {
+  SET_LOADING = 'SET_LOADING',
```
Comment (Overtorment) [inline screen/wallets/WalletsList.tsx:47]: btw, you dont have to set string values to enums, you can just list enums, and ts will assign them a sequential number under the hood

Comment (Overtorment) [reply screen/wallets/WalletsList.tsx:47]: e.g.

```ts
enum ActionTypes {
  SET_LOADING,
  SET_WALLETS,
}
```

**Review by Overtorment — CHANGES_REQUESTED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/alavtbkshpdataioljj4p2eruy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1216)

File: helpers/getIsLargeScreen.ts (hunk 1 above)
Comment (marcosrdz) [reply helpers/getIsLargeScreen.ts]: updated.

File: screen/wallets/WalletsList.tsx (hunk 2 above)
Comment (marcosrdz) [reply screen/wallets/WalletsList.tsx]: done

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/efaz5lw57cfr23bfedgv46fjw4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1223)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ixfaolifachqzw7kktzvvkairy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1226)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: helpers/getIsLargeScreen.ts (hunk 1 above)
Comment (marcosrdz) [reply helpers/getIsLargeScreen.ts]: Used to have to differentiate before. No longer need to

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/25vyqt3jf6h3pcz2r6uj3owdra?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1241)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by Overtorment — APPROVED** [review]
lgtm

**Pull request merged**